### DISCUSSION TITLE
NEXT-REDUX-WRAPPER With lots of old bugs opened with no maintainer responding

### DISCUSSION THREAD

**Discussion by Rodrigo-JM** [reduxjs/redux-toolkit#4275] [Q&A]
I have a project withe pages router I need to implement RTK and RTK Query on. 

But while you guys recommend using next-redux-wrapper for enabling SSR support, If you look at the issues, lot's of bugs with no one responding. Is this recommendation up to date?

**Comment by riqts** [comment]
The answer mey be a little bit disappointing. It's an up to date recommendation, but there is no clean resolution currently.


* Pages router is ultimately not receiving any further updates and App Router is the direction Next.js has chosen.
* Nextjs/React are amidst a big shift towards RSCs and their exact implementation is somewhat uncertain
* The approach to RTK in the App Router is a drastic change in approach to what the Pages Router and hydration supported
  * no longer treat the store like a singleton, which makes everything more testable anyway
  * use NextJS `fetch` wrapper and `cache` for server operations
  
Ultimately some SSR in the pages router can still be created using the [wrapper](https://redux-toolkit.js.org/rtk-query/usage/server-side-rendering#server-side-rendering-with-nextjs) but it is unlikely to receive much support if any, and looking forward you may have to just consider moving with Next.js :shrug:
